Userland/Libraries/LibJS/Runtime/ArrayBufferConstructor.cpp

```diff
@@ -59,7 +83,25 @@ ThrowCompletionOr<Object*> ArrayBufferConstructor::construct(FunctionObject& new
         }
         return error;
     }
-    return TRY(allocate_array_buffer(global_object(), new_target, byte_length_or_error.release_value()));
+    auto byte_length = byte_length_or_error.release_value();
+
+    // 3. Let requestedMaxByteLength be ? GetArrayBufferMaxByteLengthOption(options).
+    auto options = vm.argument(1);
+    auto requested_max_byte_length_value = TRY(get_array_buffer_max_byte_length_option(options));
+
+    // 4. If requestedMaxByteLength is empty, then
+    if (!requested_max_byte_length_value.has_value()) {
+        // a. Return ? AllocateArrayBuffer(NewTarget, byteLength).
+        return TRY(allocate_array_buffer(global_object(), new_target, byte_length));
+    }
+    auto requested_max_byte_length = requested_max_byte_length_value.release_value();
+
+    // 5. If byteLength > requestedMaxByteLength, throw a RangeError exception.
+    if (byte_length > requested_max_byte_length)
+        return vm.throw_completion<RangeError>(global_object(), ErrorType::ByteLengthBeyondRequestedMax, "array buffer");
```

Comment:
The new ErrorType `ByteLengthBeyondRequestedMax` does not have any formatting options `{}` in the string, did you mean to add one?
Otherwise just remove the `"array buffer"` from here

Reply:
Ahh that makes sense, copy-paste + not paying enough attention